tests/tests.h

```diff
@@ -32,6 +32,11 @@
 #include <sstream>
 #include <iomanip>
 
+// silence extra diagnostics in the testsuite
+#ifdef DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+#endif
```

Comment:
Since you `#include` config.h, shouldn't it always be `#define`d (possibly to the empty string)?

Also, this disables *all* diagnostics for *all* tests. Why is this necessary? I would think that there is value in these warnings, even if currently we don't do anything with them.